### PR TITLE
Fix #2652 : Improve splash screen implementation with new drawable and updated styles

### LINKED ISSUE #2652: Splash Screen Stretched
**Describe the bug**
 Splash screen is displayed when a user opens the app, on small screen devices the splash screen is working as expected, but on large screen devices, the splash screen is getting stretched and doesn't work as expected.

**To Reproduce**
1. Go to the Oppia app
2. Click on the icon
3. If your device screen is more than 6 inch
4. See  splash screen getting stretched

**Expected behavior**
 The splash screen should be displayed in the same aspect ratio as needed.

**Screenshots**

[Note: Also reported in #3193 -- see additional videos in that issue.]

Expected Behaviour
Expected Behaviour
Actual Behaviour
SplashScreen

**Device**
- OnePlus 7 Pro
- Android 10.0

The logo should be fixed as per device width:
- Mobile ([Portrait](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/ddc5589e-10d1-48e4-b551-5487801233e1), [Landscape](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/375d3300-f70d-44ed-a48a-269a58ecc854))
- Tablet ([Portrait](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/d39fdc8a-2bee-4867-a1a6-ce4280671c2b), [Landscape](https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/d69bb715-1aea-4815-8401-6e7fde167ed4))

### PR DESCRIPTION
**Pull request by TanishMoral11**
Fixes #2652: This PR aims to enhance the splash screen implementation by making the following changes:

- **splash_image.xml**: Created a new layer-list drawable that includes a background color (`@color/oppiaPrimaryLight`) and centers the splash image (`@drawable/splash_page`).
- **splash_page.xml**: Modified the path data for the splash image and removed the old clip-path to ensure proper rendering.
- **styles.xml**: Updated the `SplashScreenTheme` to reference the new `splash_image` drawable and set `adjustViewBounds` to true, allowing for better handling of different screen sizes.

These changes are intended to improve the visual consistency and responsiveness of the splash screen across various devices.

## Essential Checklist
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...").
- [x] Any changes to `scripts/assets` files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

Before : 
https://github.com/user-attachments/assets/161eb1ca-57b1-4b63-829a-de41f1b99702

After : 
https://github.com/user-attachments/assets/fcd480ff-aabd-4fe3-9887-b9bf50b276ac

### CODE REVIEW

**Comment by TanishMoral11** [comment]
@adhiamboperes PTAL

**Comment by TanishMoral11** [comment]
Hey @adhiamboperes  I Created This Pr , Can You Please Review It

File: app/src/main/res/drawable/splash_image.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+  <item android:drawable="@color/color_palette_primary_color" />
+  <item android:drawable="@drawable/splash_page" android:gravity="center"/>
+</layer-list>
```
Comment (adhiamboperes) [inline app/src/main/res/drawable/splash_image.xml]: Please insert new line at eof

**Review by adhiamboperes — APPROVED** [review]
Thanks @TanishMoral11! I just have one minor change blocking merge. PTAL.

File: app/src/main/res/drawable/splash_image.xml (hunk 1 above)
Comment (TanishMoral11) [reply app/src/main/res/drawable/splash_image.xml]: Hey @adhiamboperes  , 
I’ve added the newline at the end of `splash_image.xml` as requested. Please take a look. Thanks!

File: app/src/main/res/drawable/splash_image.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+  <item android:drawable="@color/color_palette_primary_color" />
```
Comment (adhiamboperes) [inline app/src/main/res/drawable/splash_image.xml]: @TanishMoral11, Please change this to a component colour to fix the failing check. Please refer to the closed PR for a comment I made related to this.

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
@TanishMoral11, PTAL at the issue preventing merge.

File: app/src/main/res/drawable/splash_image.xml (hunk 2 above)
Comment (TanishMoral11) [reply app/src/main/res/drawable/splash_image.xml]: Hey @adhiamboperes, Thanks for the guidance! I've updated the color to a component color as recommended to address the failing check. Please let me know if there’s anything else needed!

File: app/src/main/res/drawable/splash_image.xml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <layer-list xmlns:android="http://schemas.android.com/apk/res/android">
-  <item android:drawable="@color/color_palette_primary_color" />
+  <item android:drawable="@color/color_palette_splash_activity_background_color" />
```
Comment (adhiamboperes) [inline app/src/main/res/drawable/splash_image.xml]:
```suggestion
  <item android:drawable="@color/component_color_splash_activity_background_color" />
```

Component colors allow the app to adjust to both light and dark modes, and are located in component_colors.xml. This [link](https://github.com/oppia/oppia-android/wiki/Dark-Mode) contains more information about dark mode support at Oppia.

**Comment by adhiamboperes** [comment]
@TanishMoral11, PTAL.

File: app/src/main/res/drawable/splash_image.xml (hunk 3 above)
Comment (TanishMoral11) [reply app/src/main/res/drawable/splash_image.xml]: Hey @adhiamboperes, thanks for the suggestion! I’m updating the splash screen background to use the component color from `component_colors.xml` so it adapts to both light and dark themes. PTAL once the changes are up.

**Review by adhiamboperes — APPROVED** [review]
Thanks @TanishMoral11, this LGTM.

**Pull request merged**